### PR TITLE
Fix: set "java" platform to transactions

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :scroll: Description

Fix: set "java" platform to transactions

## :bulb: Motivation and Context

Each event/transaction that SDK sends should have platform set.

## :green_heart: How did you test it?
Unit tests.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] No breaking changes

### CODE REVIEW

**Comment by maciejwalkowiak** [comment]
The alternative implementation that I think I would be in favour of is that `SentryBaseEvent#platform` is set to "java" by default when object is created. This way we could mark it as `@NotNull` and just let user overwrite it with non-null values if needed.

If it's not something that users should be able to overwrite we could then remove the setter completely. What do you think @bruno-garcia @marandaneto

File: sentry/src/main/java/io/sentry/SentryBaseEvent.java [hunk 1]
```diff
@@ -176,4 +190,12 @@ public String getEnvironment() {
...
     this.environment = environment;
   }
+
+  public String getPlatform() {
+    return platform;
+  }
+
+  public void setPlatform(String platform) {
+    this.platform = platform;
+  }
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/SentryBaseEvent.java:200]:
```suggestion
  public @Nullable String getPlatform() {
    return platform;
  }

  public void setPlatform(final @Nullable String platform) {
    this.platform = platform;
  }
```

**Comment by marandaneto** [comment]
> The alternative implementation that I think I would be in favour of is that `SentryBaseEvent#platform` is set to "java" by default when object is created. This way we could mark it as `@NotNull` and just let user overwrite it with non-null values if needed.
> 
> If it's not something that users should be able to overwrite we could then remove the setter completely. What do you think @bruno-garcia @marandaneto

better not, we also capture events from native platforms (Android NDK), it's important to know if there was a value, or not. 
I think Gson would overwrite anyway but seems wrong.

**Comment by maciejwalkowiak** [comment]
But do we set different value for NDK? The javadocs says:

```
Acceptable values are: `as3`, `c`, `cfml`, `cocoa`, `csharp`, `elixir`, `haskell`, `go`,
   * `groovy`, `java`, `javascript`, `native`, `node`, `objc`, `other`, `perl`, `php`, `python`,
   * `ruby`
```

**Comment by marandaneto** [comment]
> But do we set different value for NDK? The javadocs says:
> 
> ```
> Acceptable values are: `as3`, `c`, `cfml`, `cocoa`, `csharp`, `elixir`, `haskell`, `go`,
>    * `groovy`, `java`, `javascript`, `native`, `node`, `objc`, `other`, `perl`, `php`, `python`,
>    * `ruby`
> ```

NDK events are `native`, Dart events are `other`, RN events are `javascript`

**Comment by maciejwalkowiak** [comment]
Ok, then PR is ready to review as it is.

**Review by marandaneto — APPROVED** [review]
LGTM

**Comment by bruno-garcia** [comment]
> But do we set different value for NDK? The javadocs says:
> 
> ```
> Acceptable values are: `as3`, `c`, `cfml`, `cocoa`, `csharp`, `elixir`, `haskell`, `go`,
>    * `groovy`, `java`, `javascript`, `native`, `node`, `objc`, `other`, `perl`, `php`, `python`,
>    * `ruby`
> ```

NDK will use `native`

**Pull request merged**